Servers/UI/OJS.Servers.Ui/ClientApp/src/components/problems/problem-resource/ProblemResource.tsx

```diff
@@ -46,24 +49,24 @@ const ProblemResource = ({ resource }: IProblemResourceProps) => {
             </Button>
         )), [ handleDownloadResourceFile, resource ]);
 
-    const resourceTypeIconClassName = useMemo(
-        () => resource.type == null
-            ? resourceTypeToIconClassName[1]
-            : resourceTypeToIconClassName[resource.type],
-        [ resource.type ],
-    );
+    const getResourceLinkContent = useCallback(() => {
+        const Icon = resourceTypeToIcon[resource.type] || null;
 
-    const resourceIconClassName = useMemo(
-        () => concatClassNames(styles.icon, resourceTypeIconClassName),
-        [ resourceTypeIconClassName ],
-    );
+        if (isNil(Icon)) {
```

Comment:
When this check is here, we will not be rendering anything through this function when the icon is null. The check should be moved in the return where the rendering of the icon is, so that we show the name of the resource at least.